mlflow/utils/model_utils.py

```diff
@@ -118,6 +119,17 @@ def _validate_and_copy_code_paths(code_paths, path, default_subpath="code"):
 
 def _add_code_to_system_path(code_path):
     sys.path = [code_path] + _get_code_dirs(code_path) + sys.path
+    # delete code modules that are already in sys.modules
+    # so they will get reloaded anew from the correct code path
+    # othterwise python will use the alredy loaded modules
+    modules = []
+    for path in [code_path] + _get_code_dirs(code_path):
+        modules_py = glob.glob(os.path.join(path, "*.py"))
+        modules += [
+            os.path.basename(f)[:-3] for f in modules_py if os.path.isfile(f) and not f.endswith("__init__.py")
+        ]
```

Comment:
We can simplify this line using `pathlib`:

```suggestion
        modules += [p.stem for p in Path(path).rglob("*.py") if p.is_file() and p.name != "__init__.py"]
```